### PR TITLE
migrate wxBlog from Blogger to official website

### PR DESCRIPTION
**Pull request by tierra**
This adds a new collection for Developer Blog posts, which remain
completely separate from news posts, and doesn't change news post
paths or affect templates that use "site.posts".

All developer blog posts are maintained in the "_blog" directory,
not mixed with the "_posts" directory for news posts. Authoring new
blog posts is mostly the same as writing news posts otherwise.

Notable Pages:

- /blog/ - Latest 3 posts, like front page news, with Archive link.
- /blog/archive/ - All blog posts, like News archive page.

A second Atom RSS feed has been added (as meta for all pages on the
site), so feed readers can pick up and subscribe to news and/or blog
separately.

All post content (including images) has been migrated inline, so
nothing is hotlinked in posts. The old wxBlog could be shut down
permanently without breaking content migrated here (but it'd be good
to leave up with a notice directing users to the new home for a year
or two).

### CODE REVIEW

**Comment by tierra** [comment]
This took me a lot longer than I expected, but it's finally finished. The Jekyll Blogger importer pretty cleanly handled file creation and YAML front matter, but left HTML untouched. So I was required to go through all posts using a combination of [turndown](https://domchristie.github.io/turndown/) and [pandoc](https://pandoc.org/) in order to convert the HTML to our much more friendly Markdown.

Every single post went through hand touchups, in many cases, fixing links, formatting, and even using much more slick tables (check out "Buzzwords Benchmarking" post for example).

Tags were imported successfully, and they are present in source Markdown, but the design just doesn't expose them anywhere. I'm not terribly sure how helpful they are, but we can certainly expose those in the UI at any point down the road, and all old posts will make use of original tags.

I've confirmed that every post on Blogger made it over into this patch.

**Comment by doscott** [comment]
It worked flawlessly for me. Looks great!

**Comment by vadz** [comment]
No real comments, looks great to me, thanks! I have just one question: do I understand correctly that any images or other link targets from a blog post `YYYY-MM-DD-xxx.md` need to be put under `blog/YYYY/MM/xxx` directory? If so, it might be nice to mention this somewhere, i.e. either in the README or maybe some new `README-BLOG` or whatever.

Would you consider a brief note about moving from Blogger on the old site when you merge it? I could do it myself, of course, but I thought it would be more appropriate if you did it.

Thanks again!

**Comment by vadz** [comment]
Oops, another question: do you know if it's possible to highlight Markdown code blocks as C++ without indenting them? I did something like this:
```
This is just a test of the **new** `wxBlog`!

Here is some code:

{% highlight cpp %}
        // Do you see the missing newline?
        printf("Hello, %s", "world");
{% endhighlight %}

Of course, you should use [wxPrintf](http://docs.wxwidgets.org) instead.
```
for testing and it works just fine, but the indent is preserved in the output. Of course, I could remove the indent -- but then the code is not highlighted in my editor. This is hardly a fatal problem, but, if possible, I'd like to have my cake and eat it too...

**Comment by vadz** [comment]
And yet another one (I think we really need a short README about adding blog posts, and I can even write it myself once I know how to do it): should we continue to use `tags:` field in the new blog posts header? I.e. are they useful for anything? I don't see them in the generated HTML and it doesn't seem possible to select all posts with the given tag (which was the main reason I used them in the first place).

**Comment by tierra** [comment]
> do I understand correctly that any images or other link targets from a blog post YYYY-MM-DD-xxx.md need to be put under blog/YYYY/MM/xxx directory?

It's not actually required, but it does help make it easier to reference in the post content, and also easier to identify which files belong with which posts. You could just save images and other content to `blog/` parent directory, and reference it in the post like:

```
<img src="../../../some-file.png">
```

I agree though, probably worth documenting my approach as just a good convention to follow.

**Comment by tierra** [comment]
> Would you consider a brief note about moving from Blogger on the old site when you merge it? I could do it myself, of course, but I thought it would be more appropriate if you did it.

Yeah, I can certainly do that.

**Comment by tierra** [comment]
> Oops, another question: do you know if it's possible to highlight Markdown code blocks as C++ without indenting them?

Yeah, my editor (Atom) doesn't know what to do about this either, as it's expecting strictly Markdown. Depending on the editor though, it might understand GFM (GitHub Flavored Markdown), and I believe this works for code highlighting as well:

    ```cpp
    // Do you see the missing newline?
    printf("Hello, %s", "world");
    ```

I didn't try it myself yet, but worth a shot.

**Comment by tierra** [comment]
> should we continue to use `tags:` field in the new blog posts header? 

Not sure if you spotted my comment about this earlier (second comment in the PR), but if you think they're helpful and have the time, it won't hurt to continue adding them to posts, even for as long as we still aren't exposing tag pages for them. I didn't build out the UI for it now, but it's certainly possible with Jekyll. Here's a [nice guide](http://longqian.me/2017/02/09/github-jekyll-tag/) on how to set that up in fact.

There's plugins to do tags too, but like that guide mentions, most (less popular) plugins aren't available on GitHub Pages, and while we're not hosting there, I've worked really hard to keep that option open, and the translation page and SSL are the only two reasons we can't right now. The former is something we can work around still, and I'm hoping GitHub fixes the latter soon with Lets Encrypt. Given those two things get resolved, I can retire that server entirely.

**Comment by vadz** [comment]
Thanks, using GFM with

    ```cpp

works perfectly for me and my editor (Vim) has no trouble with it (and can even use C++ syntax for highlighting the snippets), I didn't even think of trying this as I saw the alternative syntax being used in the existing files and thought this was how it was supposed to be done.

Also, sorry for somehow skipping over your comment about the tags, I've indeed completely failed to notice it. I do think it would be nice to have this eventually and will continue adding them, but this is definitely not very urgent.

Please go ahead and merge it and announce the blog migration whenever it's convenient for you. I just need a good idea for the next post now...

**Comment by tierra** [comment]
Nice, yeah, I'll switch the others around to that style of highlighting. I'll also add some docs before I merge this as well.

**Comment by tierra** [comment]
I've just pushed up changes to use GFM syntax highlighting along with doubling the docs provided in the README to explain news and blog posts, and tips for Markdown editing.

File: README.md [hunk 1]
```diff
@@ -67,3 +67,83 @@ files which can be retrieved from the main site:
...
+[gfm]: https://guides.github.com/features/mastering-markdown/
+
+The following GFM features are supported:
+
+* Syntax Highlighting
+* Tables
+* Automatic linking for URLs
+* Strikethrough
+
+The use of Jeykyll means that pages don't need to be entirely just Markdown. All
```
Comment (vadz) [inline README.md]: Typo in Jekyll.

**Comment by tierra** [comment]
Oh, hey, speaking of GitHub Pages SSL support 15 days ago...
https://blog.github.com/2018-05-01-github-pages-custom-domains-https/

Unrelated to this pull, but good news regardless :)

**Comment by vadz** [comment]
@tierra Should we merge this or is there anything still preventing it from happening? TIA!

**Comment by tierra** [comment]
I haven't drafted up a blog post to wrap this up just yet, but I'll try to find some time soon to do it.

**Pull request merged**